modules/oracle/src/main/scala/com/dimafeng/testcontainers/OracleContainer.scala

```diff
@@ -0,0 +1,73 @@
+package com.dimafeng.testcontainers
+
+import com.dimafeng.testcontainers.{ContainerDef, SingleContainer}
+import org.testcontainers.containers.{OracleContainer => JavaOracleContainer}
+
+class OracleContainer(dockerImageNameOverride: Option[String] = None,
+                      databaseName: Option[String] = None,
+                      oraUsername: Option[String] = None,
+                      oraPassword: Option[String] = None,
+                      containerSharedMemory: Long = OracleContainer.defaultSharedMemory)
+  extends SingleContainer[JavaOracleContainer] {
+
+  override val container: JavaOracleContainer = dockerImageNameOverride match {
+    case Some(imageNameOverride) =>
+      new JavaOracleContainer(imageNameOverride)
+    case None =>
+      new JavaOracleContainer()
+  }
+
+  container.withSharedMemorySize(containerSharedMemory)
+  databaseName.foreach(container.withDatabaseName)
+  oraUsername.foreach(container.withUsername)
+  oraPassword.foreach(container.withPassword)
+
+  def driverClassName: String = container.getDriverClassName
+
+  def jdbcUrl: String = container.getJdbcUrl
+
+  def username: String = container.getUsername
+
+  def password: String = container.getPassword
+
+  def testQueryString: String = container.getTestQueryString
+}
+
+object OracleContainer {
+
+  val defaultDockerImageName = s"oracleinanutshell/oracle-xe-11g"
+  val defaultDatabaseName    = "xe"
+  val defaultUsername        = "system"
+  val defaultPassword        = "oracle"
+  val defaultSharedMemory    = 10240000000L //1 GB
+
+  def apply(dockerImageNameOverride: String = null,
```

Comment:
If you have defaults you don't need to create extra `apply` method. Just remove `Option`s from the default constructor and use these defaults there. Like in `Def`.

Reply:
I just did the same as in the PostgresContainer! I will apply your comments

Reply:
Yes, currently we have some redundancy and inconsistency in the current codebase. I hope someday we will fix it.

Reply:
These `apply`'s are used to remove extra `new` when create a new container. I still think this should exist for each container for consistency and convince.

Reply:
Should I put it back then? I do not mind if you both agree.

Reply:
Oh, I forget that this is a simple class. I think case class can be used here.

Reply:
Oh yeah, case class would work too. The only other thing was that the companion object api exposes optionality via nullable values with default `null` so it's not needed to to wrap each value in `Option` when you use custom values.

Reply:
And this is exactly why I prefer default values instead of a bunch of options + custom `apply` with nulls.

Actually, I think this discussion is not related to the current pull request. It's more about housekeeping of the project. I think this pull request could be merged in the current state, without changes. Or, with a minimal one: turn simple class to case class.

Reply:
@miguelpuyol could you please just make it a case class?

Reply:
I will do so!